tendrl/node_agent/collectd/collectors/gluster/utils.py

```diff
@@ -0,0 +1,273 @@
+import collectd
+import ConfigParser
+import os
+import shlex
+import socket
+import subprocess
+from subprocess import Popen
+import time
+import traceback
+import ini2json
+
+
+GLUSTERD_ERROR_MSG = 'Connection failed. Please check if gluster daemon is operational.'
+
+
+def get_brick_state_mapping(status):
+    return {
+        'Started': 0,
+        'Stopped': 2
+    }.get(status, 1)
+
+
+def get_volume_state_mapping(status):
+    return {
+        'Started': 0,
+        'Stopped': 2
+    }.get(status, 1)
+
+
+def exec_command(cmd_str):
+    try:
+        cmd = Popen(
+            shlex.split(
+                cmd_str
+            ),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            stdin=open(os.devnull, "r"),
+            close_fds=True
+        )
+        stdout, stderr = cmd.communicate()
+        if stderr:
+            return None, stderr
+        return stdout, None
+    except (subprocess.CalledProcessError, ValueError):
+        except_traceback = traceback.format_exc()
+        return None, except_traceback
+
+
+def get_gluster_state_dump():
+    global GLUSTERD_ERROR_MSG
+    ret_val = {}
+    try:
+        gluster_state_dump_op, gluster_state_dump_err = exec_command(
+            'gluster get-state glusterd odir /var/run file glusterd-state'
+        )
+        if (
+            gluster_state_dump_err or
+            GLUSTERD_ERROR_MSG in gluster_state_dump_op
+        ):
+            return ret_val, gluster_state_dump_err
+        gluster_state_dump = ini2json.ini_to_dict(
```

Comment:
gluster-integration also uses ini2json, and it is used here.

Please put ini2json in commons